frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx

```diff
@@ -33,7 +33,20 @@ const TopologySideBar: React.FC<TopologySideBarProps> = ({ item, show, onClose }
     );
     const routes = metadataUIDCheck(item.resources.filter((o) => o.kind === 'Route'));
     const services = metadataUIDCheck(item.resources.filter((o) => o.kind === 'Service'));
-    const buildConfigs = metadataUIDCheck(item.resources.filter((o) => o.kind === 'BuildConfig'));
+    const builds = metadataUIDCheck(item.resources.filter((o) => o.kind === 'Builds'));
+    const buildConfigs = metadataUIDCheck(
+      item.resources.filter((o) => o.kind === 'BuildConfig'),
+    ).map((bc) => {
```

Comment:
Why is this `map` necessary?

Reply:
If there are multiple build configs then to add the builds inside the build config object based on there owner reference.
The structure would be something like this
```
buildconfigs: {
   builds: [],
   metadata: {}
   .
   .
}
```

Reply:
Should this be done here? or in topology-utils?

Reply:
@christianvogt fixed! moved the logic for builds to topology-utils.